File: Level 2/Object-Oriented-Programming-January2015/06.EncapsulationAndPolymorphism/02.Bank/Customer.cs
using System;

public abstract class Customer : ICustomer
{
    private string name;
    private int age;

    public string Name
    {
        get
        {
            return this.name;
        }

        set
        {
            if (String.IsNullOrEmpty(value))
            {
                throw new ArgumentException();
            }
            this.name = value;
        }
    }

    public int Age
    {
        get
        {
            return this.age;
        }
        set
        {
            if (value < 0)
            {
                throw new ArgumentException();
            }
            this.age = value;
        }
    }

    public Customer(string name, int age)
    {
        this.Name = name;
        this.Age = age;
    }
}

File: Level 2/Object-Oriented-Programming-January2015/08.OOP-In-Java/src/_02_Shop/Customer.java
package _02_Shop;

public class Customer {
	private String name;
	private int age;
	private double balance;
	
	public Customer(String name, int age, double balance) {
		this.setName(name);
		this.setAge(age);
		this.setBalance(balance);
	}
	
	public String getName() {
		return name;
	}
	
	private void setName(String name) {
		if (name.equals(null) || name.equals("")) {
			throw new IllegalArgumentException();
		}
		this.name = name;
	}
	
	public int getAge() {
		return age;
	}
	
	private void setAge(int age) {
		if (age < 0) {
			throw new IllegalArgumentException();
		}
		this.age = age;
	}
	
	public double getBalance() {
		return balance;
	}
	
	public void setBalance(double balance) {
		if (balance < 0) {
			throw new IllegalArgumentException("The balance cannot be negative");
		}
		this.balance = balance;
	}
}
